sql/src/test/java/io/crate/integrationtests/SQLHttpIntegrationTest.java

```diff
@@ -64,16 +67,20 @@ public void setup() {
         httpPost = new HttpPost(String.format(Locale.ENGLISH, "http://%s:%s/_sql?error_trace", address.getHostName(), address.getPort()));
     }
 
-    protected CloseableHttpClient httpClient = HttpClients.createDefault();
 
-    protected CloseableHttpResponse post(String body) throws IOException {
-        if(body != null){
-            StringEntity bodyEntity = new StringEntity(body);
+    protected CloseableHttpResponse post(String body, @Nullable Header[] headers) throws IOException {
+        if (body != null) {
+            StringEntity bodyEntity = new StringEntity(body, ContentType.APPLICATION_JSON);
             httpPost.setEntity(bodyEntity);
+            httpPost.setHeaders(headers);
```

Comment:
couldn't the header be set even if the body is null?
and does it makes sense to execute without a body?